adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -873,10 +865,11 @@
         function checkLoginIframe() {
             var promise = createPromise();
 
-            if (loginIframe.iframe && loginIframe.iframeOrigin) {
+            if (loginIframe.iframe && loginIframe.iframeOrigin && loginIframe.callbackList.length !== 0) {
                 var msg = {};
                 msg.callbackId = createCallbackId();
-                loginIframe.callbackMap[msg.callbackId] = promise;
+                msg.sessionId = kc.sessionId;
+                loginIframe.callbackList.push(promise);
```

Comment:
The current check on L876 is when the iframe is not enabled. Then it just resolves to true so it can continue. Actually you should move "loginIframe.callbackList.length !== 0" check within the current if. If iframe is enabled it should add to the list, then if the list was empty send the message. Otherwise it'll get resolved when the next message callback is received.